experimental/PropertyDDS/packages/property-binder/src/data_binder/data_binding.js

```diff
@@ -1734,8 +1734,24 @@ class DataBinding {
    * @hidden
    */
   static _registerOnValues(in_path, in_events, in_callback, in_options = {}) {
-    this.registerOnProperty(in_path, in_events, function(property) {
-      in_callback.call(this, property.isPrimitiveType() ? property.getValue() : property.getValues());
+    this.registerOnProperty(in_path, in_events, function(propertyOrKey, parent) {
+
+      // propertyOrIndex could be a primitive value when registering callbacks on primitive collection,
+      // then it value represent the index/key of the entry involved in the registered event.
+      if (propertyOrKey instanceof BaseProperty ) {
+        const property = propertyOrKey;
+        const values = propertyOrKey.isPrimitiveType() ? propertyOrKey.getValue() : propertyOrKey.getValues();
+        in_callback.call(this, values);
+      } else {
+        const key = propertyOrKey;
+        let values;
+        // parent won't be available upon removal
```

Comment:
This means the interface would be different for properties and primitive properties? In one case you get the old value and in the other case you get undefined? Should we make this consistent by always giving undefined, or is the old value valuable enough to justify the difference in the interface?

Reply:
Yeah, I think we need to revisit this logic and make it consistent.